src/App.tsx

```diff
@@ -16,47 +19,103 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [goodsArr, setGoodsArr] = useState(goodsFromServer);
+  const [sortValue, setSortValue] = useState('');
+  const [reverseValue, setReverseValue] = useState(false);
+  const [resetValue, setResetValue] = useState(false);
+
+  const sortBy = (value: string) => {
+    if (value === variables.sort_by_alphabetically) {
+      if (reverseValue) {
+        setGoodsArr(prevState => [...prevState].sort((a, b) => b.localeCompare(a)));
+      } else {
+        setGoodsArr(prevState => [...prevState].sort((a, b) => a.localeCompare(b)));
+      }
+
+      setSortValue(variables.sort_by_alphabetically);
+      setResetValue(true);
+
+      return;
+    }
+
+    if (reverseValue) {
+      setGoodsArr(prevState => [...prevState].sort((a, b) => b.length - a.length));
+    } else {
+      setGoodsArr(prevState => [...prevState].sort((a, b) => a.length - b.length));
+    }
+
+    setSortValue(variables.sort_by_length);
+    setResetValue(true);
+  };
+
+  const reset = () => {
+    setGoodsArr(goodsFromServer);
+    setReverseValue(false);
+    setResetValue(false);
+    setSortValue('');
+  };
+
+  const reverse = () => {
+    if (!sortValue) {
+      setResetValue(!resetValue);
+    } else {
+      setResetValue(true);
+    }
+
+    setGoodsArr(prevState => [...prevState].reverse());
+    setReverseValue(!reverseValue);
+  };
```

Comment:
```suggestion
  const reverse = () => setIsReversed(prevIsReversed => !prevIsReversed);
```
click on reverse button should only change isReversed state on opposite